plugins/instagram.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+import json
+import re
+from typing import List, Union
+
+import httpx
+from yt_dlp import YoutubeDL
+
+from core.exceptions import PluginError
+from core.model import PodcastFeed, PodcastItem
+from core.plugin.plugin import Plugin
+from core.utils import safe_traverse
+
+
+class PluginImpl(Plugin):
+    def get_feed(self, feed_id):
+        """Calculates and returns the subscribable feed."""
+        with httpx.Client() as client:
+            url = f'https://www.instagram.com/{feed_id}/'
+            response = client.get(url)
+            match = re.search(r'"customHeaders":({.*?}),', response.text, re.DOTALL)
+            custom_headers = match and json.loads(match.group(1))
+            api_response = client.get(
+                url=f'https://www.instagram.com/api/v1/users/web_profile_info/?username={feed_id}',
+                headers={
+                    'X-IG-App-ID': custom_headers['X-IG-App-ID']
```

Comment:
This is the bare basic for the API to work. But currently the anti bot is triggered after refreshing the feed around 15 times. We need to make the request more realistic or at some point use a library that implements this for us.